csv/result.go

```diff
@@ -296,6 +297,18 @@ func readMetadata(r *csv.Reader, c ResultDecoderConfig, extraLine []string) (tab
 		if n != len(line) {
 			return tableMetadata{}, errors.Wrap(csv.ErrFieldCount, "failed to read annotations")
 		}
+		if len(line) > 0 && line[0] == errorAnnotation {
```

Comment:
This will not work when annotations are enabled, which is expected when decoding.

Reply:
I'm not really sure I understand why this won't work. I also don't see a way to turn off annotations to begin with. Can you give me an example of what you mean so I can add it to the tests?

Reply:
Found it in the spec.

Reply:
Yes, the decoder needs the annotation in order to completely decode a result. As such the decoder assumes they exist. 

So instead of testing with:

```
error,reference
msg,ref
```

The test needs to be like:

```
#datatype,string,string
#group,false,false
#default,,
,error,reference
,msg,ref
```

I just wrote that by hand so it may have a typo, but that is the general idea.